IO/src/main/java/io/deephaven/io/streams/ByteBufferOutputStream.java

```diff
@@ -283,51 +249,45 @@ public void writeUTF(CharSequence str) throws IOException {
     // -----------------------------------------------------------------------------------
 
     public ByteBufferOutputStream appendByteBuffer(ByteBuffer bb) throws IOException {
-        if (buf.remaining() < bb.remaining()) {
-            buf = sink.acceptBuffer(buf, bb.remaining());
+        final int origPos = bb.position();
+        final int origLimit = bb.limit();
+        int remaining;
+        while ((remaining = buf.remaining()) < bb.remaining()) {
+            buf.put(bb.limit(bb.position() + remaining));
+            bb.limit(origLimit);
+            updateBuffer(buf.capacity());
```

Comment:
Is this right? Or do we want `bb.remaining()`? I guess that could blow up if we ask for too much.

Reply:
The whole idea is that we want to chunk it out instead of requesting huge buffers... the code I'm trying to update here asks for one big buffer, while the rest of this class takes the streaming approach (that I've updated this code path to here).